sale_order_project/models/sale.py

```diff
@@ -59,8 +65,34 @@ def action_create_project(self):
         project_obj = self.env['project.project']
         for order in self:
             vals = self._prepare_project_vals(order)
-            project = project_obj.create(vals)
-            order.write({
-                'project_id': project.analytic_account_id.id
-            })
-        return True
+            if order.project_template_id:
+                new_date_start = time.strftime('%Y-%m-%d')
+                new_date_end = False
+                if order.project_template_id.date_start and \
+                        order.project_template_id.date:
+                    start_date = date(*time.strptime(
+                            order.project_template_id.date_start, '%Y-%m-%d'
+                        )[:3])
+                    end_date = date(*time.strptime(
+                            order.project_template_id.date,
+                            '%Y-%m-%d')[:3])
+                    new_date_end = (datetime(*time.strptime(
+                            new_date_start,
+                            '%Y-%m-%d'
+                        )[:3])+(end_date-start_date)).strftime('%Y-%m-%d')
+                vals.update({
+                        'state': 'open',
+                        'date_start': new_date_start,
+                        'date': new_date_end,
+                        'parent_id': order.project_template_id.parent_id.id
+                    })
+                new_id = order.project_template_id.copy(default=vals)
+                order.write({
+                    'project_id': new_id.analytic_account_id.id
+                })
+            else:
+                project = project_obj.create(vals)
+                order.write({
```

Comment:
The `order.write` is repeated in the if and else. You can have it only once, outside the if-else.

Reply:
Done, seems github doesn't take the change as done but it's out of the if/else now. Thanks for pointing it out!